Dockerfile

```diff
@@ -197,7 +197,7 @@ RUN git init . && git remote add origin "https://github.com/containerd/container
 # When updating the binary version you may also need to update the vendor
 # version to pick up bug fixes or new APIs, however, usually the Go packages
 # are built from a commit from the master branch.
-ARG CONTAINERD_VERSION=v1.7.22
+ARG CONTAINERD_VERSION=v2.0.0
```

Comment:
We should probably also have a run of CI with the v1.7 binaries, but let's see what v2.0.0 says first

Reply:
Is CI typically run with multiple versions?

Do you have https://github.com/moby/moby/pull/47335 just for testing?

Reply:
We don't run with multiple versions, as we're already rate-limited by GitHub actions for too many jobs, but we should probably have at least a run of CI with either combination; I opened https://github.com/moby/moby/pull/47335 to verify if our existing code would work fine with containerd 2.0, because the moment we push packages of containerd 2.0, every installation will use it, including older versions of docker engine (unless people explicitly pin to 1.7).

The reverse may also be true; a user may upgrade docker and potentially use an 1.7 version of containerd; we can restrict that in our packaging, but it might be good to know if it works regardless.